src/org/labkey/test/tests/ScriptValidationTest.java

```diff
@@ -147,15 +148,16 @@ public void doTestCrossFolderSaveRows() throws Exception
                 new Object[][]{new Object[]{modelId, colors.get(0).name, 2000, 1234, new Date()}}, null);
         CommandResponse response = saveRowsCommand.execute(cn, "/home");
         Map<String, Object> row = (Map)((Map)((List)((Map)((List)response.getParsedData().get("result")).get(0)).get("rows")).get(0)).get("values");
-        Object vehicleRowId = row.get("rowid");
+        Object vehicleRowId = row.get("rowId");
```

Comment:
Consider using a case-insensitive map
```suggestion
        Object vehicleRowId = new CaseInsensitiveHashMap(row).get("rowId");
```
Other references to "rowId" in this test should be ok, since they are inspecting `SelectRowsResponse`, which already uses such a map. (no harm in updating them though)

Reply:
Generally a good idea, but labkey should be deterministic in the case in which it returns columns. While our test could be case-insensitive, JS code is case sensitive and changing around the server's response is a big deal.

I think something has changed in the case in which columns are returned. I would have expected the response to mirror the case in vehicles.xml, and the schema browser shows this. So why is the SaveRowsResponse converting everything to lowerCamelCase?

Reply:
Yes, this is weird. We're proving SaveRowsAction with a map using the camel case values (which matches what shows as the field case in schema browser:

![image](https://user-images.githubusercontent.com/7724863/209716272-8694add9-dd04-4cb8-8e8c-2f45be652ffa.png)


here's the actual map in java (rowsToProcess):

![image](https://user-images.githubusercontent.com/7724863/209716171-64561023-4258-4571-a299-4966b13f9430.png)

however, after the call to: 

List<Map<String, Object>> responseRows =
                        commandType.saveRows(qus, rowsToProcess, getUser(), container, configParameters, extraContext);

The case on responseRows has all been normalized to lowerCamelCase:

![image](https://user-images.githubusercontent.com/7724863/209715987-7c697b42-829f-4b1b-8936-1d003383fe6f.png)

Reply:
Update: the row's case is preserved until QueryController line 3989, where it takes the updated rows and re-selects them. This re-selected map has lowerCamelCase keys.

```
                // Issue 42519: Submitter role not able to insert
                // as per the definition of submitter, should allow insert without read
                if (qus.hasPermission(user, ReadPermission.class))
                {
                    updatedRows = qus.getRows(user, container, updatedRows); 
                }
```

It maps further into DefaultQueryUpdateService._select, line 233. The original case is preserved until here. TableSelector is called using the PK, and the resulting map has the field keys using lowerCamelCase. However, I cannot figure out any change that plausibly changed how this behavior worked though.